packages/core/src/app/customer/Customer.tsx

```diff
@@ -458,11 +487,21 @@ export function mapToWithCheckoutCustomerProps({
         },
     } = checkoutState;
 
+    const cart = getCart();
     const billingAddress = getBillingAddress();
     const checkout = getCheckout();
     const customer = getCustomer();
     const signInEmail = getSignInEmail();
     const config = getConfig();
+    let stripeUpeLinkEnabled = false;
+
+    if (cart) {
+        const stripeUpe = getPaymentMethod('card', PaymentMethodId.StripeUPE);
+        const linkEnabled = stripeUpe?.initializationData.enableLink || false;
+        const stripeUpeSupportedCurrency = cart.currency.code === 'USD' || false;
+
+        stripeUpeLinkEnabled = linkEnabled && stripeUpeSupportedCurrency;
+    }
```

Comment:
@BC-AOrtiz we need to add another TODO here, this should be driven by backend, same as other wallet buttons.